src/postgres-util/src/lib.rs

```diff
@@ -232,6 +233,8 @@ pub async fn connect_replication(conn: &str) -> Result<Client, anyhow::Error> {
     let tls = make_tls(&config)?;
     let (client, connection) = config
         .replication_mode(ReplicationMode::Logical)
+        .connect_timeout(Duration::from_secs(30))
+        .keepalives_idle(Duration::from_secs(10 * 60))
```

Comment:
the default connection timeout is unbounded, now 30s. timing out should result in a retry

the default keepalive idle (when a TCP connection starts to issue keep-alives after being idle) is 2 hours, now 10 minutes. I don't imagine this one frequently coming up.